### ISSUE TITLE
Mixed Precision Training

### ISSUE THREAD

**Issue by Carapuzowski** [Carapuzowski/functions#3]
Write a Python class to implement Mixed Precision Training that uses both float32 and float16 data types to optimize memory usage and speed. Your class should have an __init__(self, loss_scale=1024.0) method to initialize with loss scaling factor. Implement forward(self, weights, inputs, targets) to perform forward pass with float16 computation and return Mean Squared Error (MSE) loss (scaled) in float32, and backward(self, gradients) to unscale gradients and check for overflow. Use float16 for computations but float32 for gradient accumulation. Return gradients as float32 and set them to zero if overflow is detected. Only use NumPy.

**Comment by Carapuzowski** [comment]
Example:
```python
import numpy as np
mp = MixedPrecision(loss_scale=1024.0)
weights = np.array([0.5, -0.3], dtype=np.float32)
inputs = np.array([[1.0, 2.0], [3.0, 4.0]], dtype=np.float32)
targets = np.array([1.0, 0.0], dtype=np.float32)
loss = mp.forward(weights, inputs, targets)
print(f"Loss: {loss:.4f}")
print(f"Loss dtype: {type(loss).__name__}")
grads = np.array([512.0, -256.0], dtype=np.float32)
result = mp.backward(grads)
print(f"Gradients: {result}")
print(f"Grad dtype: {result.dtype}")
```
